src/t8_forest/t8_forest_netcdf.cxx

```diff
@@ -0,0 +1,1188 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/*
+Description:
+These functions write a file in the NetCDF-format which represents the given 2D- or 3D-forest
+*/
+
+#include <t8.h>
+#if T8_WITH_NETCDF
+#include <netcdf.h>
+/* do we need to question wheter this header is accessible? */
```

Comment:
A configuration macro was added (`T8_WITH_NETCDF_PAR`) which automatically checks for the accessibility of the parallel netCDF routines (but only in case t8code is configured with the option `--with-netcdf`, otherwise it is `0`). (To force a serial netCDF usage, although the parallel routines are accessible, one can use `--with-netcdf=serial`)